### PR TITLE
Upgrade to JRuby 9k

### PR DESCRIPTION
**Pull request by ph**
This is a joint effort between @guyboertje  and @ph; this is the minimal changes required to make Logstash work under JRuby 9k.

### CODE REVIEW

File: rakelib/artifacts.rake [hunk 1]
```diff
@@ -1,5 +1,20 @@
 require "logstash/version"
 
+
```
Comment (suyograo) [inline rakelib/artifacts.rake]: Mistake?

Comment (ph) [inline rakelib/artifacts.rake]: /facepalm.

**Comment by ph** [comment]
I've look at the test the offline pack is currently failling and I can reproduce the error everytime locally. This is due to the fact that we rely on Rubygems' library to handle the indexes and 9k ships with a newer version of it that break the current code.

File: rakelib/childprocess_patch.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+      def ignore_close_io
+        old_pump
+      rescue IOError
+      end
+
+      alias_method :pump, :ignore_close_io
+    end
+  end
+end
```
Comment (ph) [inline rakelib/childprocess_patch.rb]: This might need to go in FPM.

**Comment by ph** [comment]
@guyboertje any ideas about theses  message: `jruby: warning: unknown property jruby.cext.enabled` in the travis job? It like something is set by default in the Travis environment.

**Comment by guyboertje** [comment]
LGTM

**Comment by ph** [comment]
squashed, waiting on ci, will merge after.

**Comment by ph** [comment]
[master 76286b4f0]

**Pull request closed without merging**